app/src/main/java/org/mozilla/fenix/tabstray/syncedtabs/SyncedTabs.kt

```diff
@@ -0,0 +1,352 @@
+ * @param errorButton Optional class to set up and handle any clicks in the Error UI.
+ */
+@Composable
+fun SyncedTabsErrorItem(errorText: String, errorButton: SyncedTabsListItem.ErrorButton? = null) {
+    Box(
+        Modifier
+            .padding(all = 16.dp)
+            .height(IntrinsicSize.Min)
+    ) {
+        val dashColor = FirefoxTheme.colors.borderPrimary
+
+        Canvas(Modifier.fillMaxSize()) {
+            drawRoundRect(
+                color = dashColor,
+                style = Stroke(
+                    width = 2.dp.toPx(),
+                    pathEffect = PathEffect.dashPathEffect(floatArrayOf(4.dp.toPx(), 4.dp.toPx()), 0f)
+                ),
+                cornerRadius = CornerRadius(
+                    x = 8.dp.toPx(),
+                    y = 8.dp.toPx()
+                ),
+            )
+        }
+
+        Column(
+            Modifier
+                .padding(all = 16.dp)
+                .fillMaxWidth()
+        ) {
+            PrimaryText(
+                text = errorText,
+                modifier = Modifier.fillMaxWidth(),
+                fontSize = 14.sp
+            )
+
+            errorButton?.let {
+                Spacer(modifier = Modifier.height(12.dp))
+
+                SyncedTabsErrorButton(buttonText = it.buttonText, onClick = it.onClick)
+            }
+        }
+    }
+}
+
+/**
+ * Error button UI within SyncedTabsErrorItem
+ *
+ * @param buttonText The error button's text and accessibility hint.
+ * @param onClick The lambda called when the button is clicked.
+ */
+@Composable
+fun SyncedTabsErrorButton(buttonText: String, onClick: () -> Unit) {
+    Button(
+        onClick = onClick,
+        modifier = Modifier.clip(RoundedCornerShape(size = 4.dp)),
+        elevation = ButtonDefaults.elevation(defaultElevation = 0.dp, pressedElevation = 0.dp),
+        colors = ButtonDefaults.outlinedButtonColors(backgroundColor = FirefoxTheme.colors.actionPrimary),
+    ) {
+        Icon(
+            painter = painterResource(R.drawable.ic_sign_in),
+            contentDescription = null,
+            tint = FirefoxTheme.colors.textOnColor,
+        )
+
+        Spacer(Modifier.width(8.dp))
+
+        Text(
+            text = buttonText,
+            modifier = Modifier.align(Alignment.CenterVertically),
+            color = FirefoxTheme.colors.textOnColor,
+            fontSize = 14.sp,
+            fontFamily = FontFamily(Font(R.font.metropolis_semibold)),
+            maxLines = 2
+        )
+    }
+}
+
+/**
+ * UI to be displayed when a user's device has no synced tabs.
+ */
+@Composable
+fun SyncedTabsNoTabsItem() {
+    SecondaryText(
+        text = stringResource(R.string.synced_tabs_no_open_tabs),
+        modifier = Modifier
+            .padding(horizontal = 16.dp, vertical = 8.dp)
+            .fillMaxWidth(),
+        fontSize = 16.sp,
+        maxLines = 1
+    )
+}
+
+@Composable
+@Preview
+private fun SyncedTabsListItemsPreview() {
+    FirefoxTheme(darkTheme = true) {
```

Comment:
I think we should specify the darkTheme in the `@Preview` instead.

Reply:
This was the better solution, since it no longer re-compiles the `@Preview` when just changing from light to dark